.github/workflows/branch_preparation.yml

```diff
@@ -0,0 +1,116 @@
+name: Branch Preparation CI
+
+on:
+  push:
+    branches:
+      - 'release/**'
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+**'
+
+jobs:
+  prepare_develop_branch_after_release:
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - run: |
+          git checkout develop
```

Comment:
I wonder if here we should trigger an action from the develop code itself.
Let's image what happens if the structure of the repo changes, then these steps here might not work on develop?
Can we instead trigger an action from the develop branch instead?

Reply:
We are checking out the develop here that means the script is running from the develop branch so it should work?

Reply:
Yeah, this probably won't change very often, so should be fine?